lib/queuePopulator/IngestionPopulator.js

```diff
@@ -139,6 +161,372 @@ class IngestionPopulator {
         this._mProducer.setupProducer(cb);
     }
 
+    /**
+     * Setup the Redis Subscriber which listens for actions from other processes
+     * (i.e. BackbeatAPI for pause/resume)
+     * @param {Object} redisConfig - redis config
+     * @return {undefined}
+     */
+    _setupRedis(redisConfig) {
+        // redis pub/sub for pause/resume
+        const redis = new Redis(redisConfig);
+        // redis psubscribe and will need to parse location name
+        const { topic } = this.ingestionConfig;
+        const channelPattern = `${topic}-*`;
+        redis.psubscribe(channelPattern, err => {
+            if (err) {
+                this.log.fatal('failed to subscribe to redis channel', {
+                    method: 'IngestionPopulator._setupRedis',
+                    error: err,
+                });
+                process.exit(1);
+            }
+            this._redis = redis;
+
+            redis.on('pmessage', (channel, pattern, message) => {
```

Comment:
A stylistic point: seems a bit unusual to find the event listener nested in the subscriber.